pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -2153,9 +2159,11 @@ def allocate_budget_to_maximize_response(
             custom_constraints=custom_constraints,
         )
 
-        inverse_scaled_channel_spend = self.channel_transformer.inverse_transform(
+        inverse_scaled_channel_spend = (
             np.array([list(self.optimal_allocation_dict.values())])
+            * self.channel_transformer["scaler"].scale_.max()
```

Comment:
Is there a way to add a test to ensure this change is working as expected?

Reply:
Totally agree, just added three test to verify if the input budget and estimated budget are the same. Same thing with bounds, and periods.